pkg/proto/agentbaker/v1/config.proto

```diff
@@ -225,4 +230,40 @@ message NetworkConfig {
   string cni_plugins_url = 7;
   string subnet = 8;
   string route_table = 9;
+}
+
+message CustomLinuxOSConfig {
+  bool should_config_swap_file = 1;
+  bool should_config_transparent_hugepage = 2;
+  bool should_config_containerd_ulimits = 3;
+  string containerd_ulimits = 4;
+  string swap_file_size_mb = 5;
+  string transparent_hugepage = 6;
+  string transparent_defrag = 7;
+}
+
+message CustomSearchDomain {
+  string custom_search_domain_name = 1;
+  string custom_search_domain_realm_user = 2;
+  string custom_search_domain_realm_password = 3;
+}
+
+message TLSBootstrapping {
+  // discuss whether this should be featureState or bool
+  // understand how defaulting works for enum, because we want default to be false
+  FeatureState tls_bootstrapping = 1;
+  FeatureState secure_tls_bootstrapping = 2;
+  string tls_bootstrap_token = 3;
+}
+
+message CustomCloudConfig {
+  string target_environment = 1;
+  string target_cloud = 2;  // can probably get rid of this, analyze more
+  bool is_custom_cloud = 3; 
+}
+
+message ServicePrincipal {
```

Comment:
There is a message type IdentityConfiguration above, which consolidates SP, system/user assigned identity. I think we can merge this with that.